src/Persistence/MassTransit.EntityFrameworkCoreIntegration/Saga/EntityFrameworkSagaRepository.cs

```diff
@@ -143,15 +143,13 @@ await dbContext.Database.BeginTransactionAsync(_isolationLevel, context.Cancella
             {
                 if (instance == null)
                 {
-                    IQueryable<TSaga> queryable = QuerySagas(dbContext);
-
                     // Query with a row Lock instead using FromSql. Still a single trip to the DB (unlike EF6, which has to make one dummy call to row lock)
                     var rowLockQuery = _rawSqlLockStatements?.GetRowLockStatement<TSaga>(dbContext);
                     if (rowLockQuery != null)
-                        instance = await queryable.FromSql(rowLockQuery, new object[] {sagaId}).SingleOrDefaultAsync(context.CancellationToken)
+                        instance = await ApplyCustomQuery(dbContext, dbSet => dbSet.FromSqlRaw(rowLockQuery, new object[] {sagaId})).SingleOrDefaultAsync()
```

Comment:
don't forget the  `.SingleOrDefaultAsync(context.CancellationToken).ConfigureAwait(false)`